libs/aws/langchain_aws/chat_models/bedrock.py

```diff
@@ -619,15 +623,29 @@ def _combine_llm_outputs(self, llm_outputs: List[Optional[dict]]) -> dict:
 
     def get_num_tokens(self, text: str) -> int:
         if self._model_is_anthropic:
-            return get_num_tokens_anthropic(text)
-        else:
-            return super().get_num_tokens(text)
+            bad_deps = check_anthropic_tokens_dependencies()
+            if not bad_deps:
+                return get_num_tokens_anthropic(text)
+            else:
+                logger.debug(
+                    "Falling back to default token counting due to incompatible/missing Anthropic dependencies:"
+                )
+                for x in bad_deps:
+                    logger.debug(x)
```

Comment:
Would this cause duplicate logger messages, as `super().get_num_tokens(text)` ends up calling the `get_token_ids` method. Also, adding a logger message would just fall under the radar for most users, unless they have the debug logging switched on, should we rather use a warning?

```python
import warnings

if self._model_is_anthropic and not self.custom_get_token_ids:
    warnings.warn(
        "Falling back to default token counting due to incompatible anthropic count_tokens API. "
        "For anthropic versions > 0.38.0, it is recommended to provide a custom_get_token_ids "
        "method to the chat model class that implements the appropriate tokenizer for Anthropic."
    )
```

Reply:
Thanks for the catch - removed the warning from `get_num_tokens`, and updated the message as suggested.